code-challenges.test.js

```diff
@@ -10,18 +10,59 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
+
+describe("whatsYaOccupation", () => {
+    it("returns an array with a sentence about each person with their name capitalized", () => {
+      const people = [
+        { name: "ford prefect", occupation: "a hitchhiker" },
+        { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+        { name: "arthur dent", occupation: "a radio employee" }
+      ]
+      expect(whatsYaOccupation(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
+
+// // ReferenceError: whatsYaOccupation is not defined
+
+// // b) Create the function that makes the test pass.
+
+// Pseudo Code:
+  // Create a function named whatsYaOccupation that takes in an array of objects as an argument
+  // Input: array of objects with parameter named array
+  // Iterate over the array using HOF .map with parameter object and set to a variable called arrayOfPpl since we will be deconstructing the objects within the array
+  // Use object deconstruction to access name values and both accessor and mutator methods to capitalize the first names and last names. 
+  // object.name.charAt(0).toUpperCase to capitalize the first letter concatenated to the remainder of the letters up until the first ' ' using object.name.slice(1, object.name.indexOf(' ')) which is then concatenated to a ' '  between the first/last names then use .indexOf(' ' + 1) to find the first letter that comes after the space and capitalize the last name, then concat to remainder of string
+  // Use string interpolation to make a sentence with the capitalized name and object.occupation 
+  // Set string interpolated and capitalized sentence to a variable called sentence
+  // Use mutator method .split() to convert the sentence into an array and return it
+  // return arrayOfPpl which is an array of arrays of sentences, and use the .flat() method to convert all nested arrays into one array of 3 sentences 
+  // Output: array with a sentence about each person and their name capitalized ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+
 const people = [
   { name: "ford prefect", occupation: "a hitchhiker" },
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
   { name: "arthur dent", occupation: "a radio employee" }
 ]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
-// b) Create the function that makes the test pass.
+const whatsYaOccupation = (array) => {
+  arrayOfPpl = array.map(object => {
+    sentence = (`${(object.name.charAt(0).toUpperCase() + object.name.slice(1, object.name.indexOf(' ')) + ' ' + object.name.charAt(object.name.indexOf(' ') + 1).toUpperCase() + object.name.slice(object.name.indexOf(' ') + 2))} is ${object.occupation}.`)
```

Comment:
While Javascript has become much less strict about variable declarations, creating a Ruby vibe when creating variables, it is highly recommended and best practice to still declare the variable in Javascript. The default behavior without I believe will be much looser and can lead to bugs in larger code bases.  When in doubt, use let or const when possible.